### PR TITLE
Rewrite the KerasHub Getting Started Guide

### PR DESCRIPTION
**Pull request by mattdangerw**
I had to pin nightly a few days book, look like there was some issue with ResNet on the latest version. This also needs a change to Keras that will not yet be on nightly until tonight

### CODE REVIEW

**Comment by divyashreepathihalli** [comment]
Hi Matt!! The resizing has been added back and the new presets have been updated. So is the nightly working fine for you?

File: guides/keras_hub/getting_started.py [hunk 1]
```diff
@@ -1,633 +1,788 @@
...
+deeply customizable set of building blocks.
+"""
+
+"""
+## Going further
+
+This is just scratching the surface of what you can do with the KerasHub.
+
+This guide shows a few of the high-level tasks that we ship with the KerasHub library,
+but there are many tasks we did not cover here. Try [generating images with Stable
```
Comment (divyashreepathihalli) [inline guides/keras_hub/getting_started.py:774]: we can maybe link to this page too here to point to the full list of models to try out - https://keras.io/api/keras_hub/models/

File: guides/keras_hub/getting_started.py [hunk 2]
```diff
@@ -1,633 +1,788 @@
...
+Lastly, we need to do some extra setup to access the models used in this guide. Many
+popular open LLMs, such as Gemma from Google and Llama from Meta, require accepting
+a community license before accessing the model weights. We will be using Gemma in this
+guide, so we can follow the following steps:
 
-# Use mixed precision to speed up all training in this guide.
-keras.mixed_precision.set_global_policy("mixed_float16")
+1. Go to the [Gemma 2](https://www.kaggle.com/models/keras/gemma2) model page, and accept
+   the license at the banner at the top.
+2. Generate an Kaggle API key by going to [Kaggle settings](https://www.kaggle.com/settings)
```
Comment (divyashreepathihalli) [inline guides/keras_hub/getting_started.py:68]: NIT: a Kaggle

File: guides/keras_hub/getting_started.py [hunk 3]
```diff
@@ -1,633 +1,788 @@
...
+   and clicking "Create New Token" button under the "API" section.
+3. Inside your colab notebook, click on the key icon on the left hand toolbar. Add two
+   secrets: `KAGGLE_USERNAME` with your username, and `KAGGLE_KEY` with the API key you just
+   created. Make these secrets visible to the notebook you are running.
```
Comment (divyashreepathihalli) [inline guides/keras_hub/getting_started.py:72]: NIT: Provide access to these secrets in your notebook when prompted while running the Colab.

File: guides/keras_hub/getting_started.py [hunk 4]
```diff
@@ -1,633 +1,788 @@
...
-    return x, y
+"""
+We can load the dataset with `keras.utils.image_dataset_from_directory`. One important
+thing to note here is that the `train_ds` and `val_ds` will both be returned as
+`tf.data.Dataset` objects, including on the `torch` and `jax` backends.
 
+KerasHub will use [tf.data](https://www.tensorflow.org/guide/data) as the default API for
+running multi-threaded preprocessing on the CPU. `tf.data` is a powerful API for training
+input pipelines that can scale up to complex, multi-host training jobs easily. Using it
+does not restrict your choice of backend, a `tf.data.Dataset` can be as an iterator of
```
Comment (divyashreepathihalli) [inline guides/keras_hub/getting_started.py:398]: NIT: can be used as an iterator

**Review by divyashreepathihalli — APPROVED** [review]
Thanks Matt! left a few NIT comments.

**Review by fchollet — APPROVED** [review]

**Pull request merged**